dissect/target/tools/query.py

```diff
@@ -173,14 +173,31 @@ def main():
     if not args.function:
         parser.error("argument -f/--function is required")
 
-    # Verify uniformity of output types, otherwise default to records
+    # Verify uniformity of output types, otherwise default to records.
+    # Note that this is a heuristic, the targets are not opened yet because of
+    # performance, so it might generate a false positive
+    # (os.* on Windows includes other OS plugins),
+    # however this is highly hypothetical, most plugins across OSes have
+    # the same output types and most output types are records anyway.
+    # Furthermore we really want the notification at the top, so this is the only
+    # way forward. In the very unlikely case you have a
+    # collection of non-record plugins that have record counterparts for
+    # other OSes just refine the wildcard to exclude other OSes.
+    # The only scenario that might cause this is with
+    # custom plugins with idiosyncratic output across OS-versions/branches.
     output_types = set()
     funcs = find_plugin_functions(Target(), args.function, False)
     for func in funcs:
         output_types.add(func.output_type)
 
     default_output_type = None
-    if len(output_types) > 1:
+
+    # Otherwise you see a confusing mixed output types message, can still generate output
+    # because some plugins have output file/dir options
```

Comment:
This only applies to the first `if` and not the whole block? Maybe add these comments then after the `if`